### PR TITLE
Letterhome apis

### PR DESCRIPTION
**Pull request by charliefischer**

### CODE REVIEW

File: app/letterhome/page.js [hunk 1]
```diff
@@ -15,144 +15,127 @@ import BottomNavBar from '@/components/bottom-nav-bar';
...
+				const docRef = doc(db, "users", uid);
+				const docSnap = await getDoc(docRef);
+
+				if (docSnap.exists()) {
+					const userData = docSnap.data();
+					setUserName(userData.first_name || 'Unknown User');
+					setCountry(userData.country || 'Unknown Country');
+				} else {
+					console.log("No such document!");
+				}
```
Comment (madarez) [inline app/letterhome/page.js:42]: When someone hasn't created their user profile, we should redirect them to profile page.

File: app/letterhome/page.js [hunk 2]
```diff
@@ -15,144 +15,127 @@ import BottomNavBar from '@/components/bottom-nav-bar';
...
+				for await (const l of letters) {
+					const rec = await fetchRecipients(l.letterboxId)
+					l.recipients = rec
+				}
+				console.log(letters)
+				setLetters(letters)
+
+			} else {
+				setError('No user logged in.');
+				setIsLoading(false);
```
Comment (madarez) [inline app/letterhome/page.js:66]: Should we redirect to login page?

File: app/letterhome/page.js [hunk 3]
```diff
@@ -15,144 +15,127 @@ import BottomNavBar from '@/components/bottom-nav-bar';
...
+							<Link href="/letterhome">
+								<button className="px-3 py-1 text-white bg-blue-600 rounded-lg hover:bg-blue-700 transition-colors duration-300">Show more</button>
+							</Link>
+						</h2>
+						{letters.length > 0 ? (
+							letters.map((letter, i) => (
+								<a key={letter.id + '_' + i} href={`/letters/${letter.letterboxId}`} className="flex items-center p-4 mb-3 rounded-lg bg-white shadow-md hover:shadow-lg transition-shadow duration-300 cursor-pointer">
+									<div className="flex-grow">
+										{letter.recipients?.map(rec => (
+											<div key={rec.first_name + '_' + rec.last_name} className='flex'>
```
Comment (madarez) [inline app/letterhome/page.js:113]: What do you think that we use userId as the key here?

File: app/letterhome/page.js [hunk 4]
```diff
@@ -15,144 +15,127 @@ import BottomNavBar from '@/components/bottom-nav-bar';
...
+												<div className="w-12 h-12 relative mr-4">
+												{rec?.profile_picture ? (
+													<img src={rec?.profile_picture} class="w-full h-full object-cover" />
+												) : (
+													<span className="text-xl text-gray-600">
+														{rec?.first_name?.[0]}
+													</span>
+												)}
+												</div>
+												<h3 className="font-semibold text-gray-800">{rec.first_name} {rec.last_name}</h3>
```
Comment (madarez) [inline app/letterhome/page.js:123]: We need a draft symbol for draft messages prefix the name here as well:
Screenshot 2024-06-30 at 6 23 06 PM

File: app/utils/firestore.js [hunk 5]
```diff
@@ -24,10 +24,12 @@ export const fetchData = async () => {
 
       const sentLettersQuerySnapshot = await getDocs(
         query(lettersCollectionRef,
-          where("status", "==", 'sent'),
-          where("deleted_at", "==", null),
+          // where("status", "==", 'sent'),
+          where("content", "!=", null),
+          where("deleted", "==", null),
+          where("draft", "==", false),
```
Comment (madarez) [inline app/utils/firestore.js:30]: Remind me: are we not using `status` field anymore?

File: app/letterhome/page.js (hunk 1 above)
Comment (charliefischer) [reply app/letterhome/page.js:42]: I'm hestitant on this. I think we do this once everything else is setup. Whenever you land on a page it always registers as signed out, then after on the client side we then confirm we are signed in. May be better to have a store or a cookie where we save temporarily whether the user is signed in or not

File: app/letterhome/page.js (hunk 3 above)
Comment (charliefischer) [reply app/letterhome/page.js:112]: it doesn't exist on the recipient object currently!

Comment (madarez) [reply app/letterhome/page.js:116]: Maybe we are referring to two diff things, `userId` itself doesn't exist, but I'd be surprised if the document `id` doesn't exist. That should exist in all reference objects; no?

**Pull request merged**